src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/course/restapi/CoursesIntegrationTest.kt

```diff
@@ -2,74 +2,76 @@ package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.course.restapi
 
 import io.kotest.matchers.collections.shouldContainExactly
 import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import io.kotest.matchers.collections.shouldHaveAtLeastSize
 import io.kotest.matchers.collections.shouldHaveSize
+import io.kotest.matchers.nulls.shouldNotBeNull
 import org.hamcrest.Matchers.startsWith
 import org.junit.jupiter.api.Test
-import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.boot.test.context.SpringBootTest
 import org.springframework.context.annotation.Import
-import org.springframework.core.ParameterizedTypeReference
 import org.springframework.http.MediaType
 import org.springframework.test.annotation.DirtiesContext
 import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.web.reactive.server.EntityExchangeResult
 import org.springframework.test.web.reactive.server.WebTestClient
-import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.api.model.Course
-import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.api.model.CourseOffering
+import org.springframework.test.web.reactive.server.expectBody
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.integration.IntegrationTestBase
 import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.integration.fixture.JwtAuthHelper
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.util.MEDIA_TYPE_TEXT_CSV
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.util.generateCourseRecords
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.util.generateOfferingRecords
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.util.generatePrerequisiteRecords
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.util.toCourseCsv
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.util.toOfferingCsv
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.util.toPrerequisiteCsv
 import java.util.UUID
-import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.api.model.Course as ApiCourse
 
```

Comment:
This is a critical mistake on my part, you're right. Those generation methods should have allowed for that linkage, e.g. `generateCourseOfferings(3).withCourse(foo), or perhaps `generateCourseOfferings(amount = 3, courseId = "someUuid")`. That would allow us to set the relationship between generated entities whilst generating everything else - similar to how the entity factories work.